gravatar-quickeditor/src/main/java/com/gravatar/quickeditor/ui/editor/AuthenticationMethod.kt

```diff
@@ -0,0 +1,36 @@
+package com.gravatar.quickeditor.ui.editor
+
+import com.gravatar.quickeditor.ui.oauth.OAuthParams
+import java.util.Objects
+
+/**
+ * Represents the authentication method used for the Gravatar Quick Editor.
+ */
+public sealed class AuthenticationMethod {
```

Comment:
We have two options:
1. Overload `GravatarQuickEditorBottomSheet` with different params: `OAuthParams` or `String` (for the token), or...

2. Have one `GravatarQuickEditorBottomSheet` that takes a sealed class as a parameter. I think this one is cleaner and better represents the options. It requires less copy pasting as well.

Reply:
I think I will also place that in the `GravatarQuickEditorParams` to have less params on the main composable. WDYT?